src/main/java/seedu/address/model/student/Student.java

```diff
@@ -35,13 +35,15 @@ public class Student {
     /**
      * Every field must be present and not null.
      */
-    public Student(Name name, Phone phone, Email email, Address address, Set<Tag> tags, List<ModuleCode> modules) {
+    public Student(Name name, Phone phone, Email email, Address address, Set<Tag> tags, List<ModuleCode> modules,
+                   List<ModuleTiming> moduleTimings) {
         requireAllNonNull(name, phone, email, address, tags);
```

Comment:
Would it be useful to instead have a `Module` that encapsulates `ModuleCode` and `ModuleTiming`. I think this approach might make more sense since the student object does not need to interact with `ModuleCode` and `ModuleTiming` directly, and can instead operate at a higher level of abstraction with just a list for `Module`.

We can consider doing this as a quality improvement in the next iteration!

Reply:
I was thinking about that too but I realised that we only instantiate each `Module` once and store them all inside the `ModuleMap`, which was why I didn't put `ModuleTiming` under `Module`.